service/src/main/kotlin/app/cash/backfila/service/runner/BackfillRunner.kt

```diff
@@ -65,37 +66,39 @@ class BackfillRunner private constructor(
   }
 
   fun run() {
-    logger.info { "Runner starting: ${logLabel()} " }
+    factory.loggingSetupProvider.withLogging(backfillName, backfillRunId, partitionName) {
+      logger.info { "Runner starting: ${logLabel()} " }
 
-    metadata = factory.transacter.transaction { session -> loadMetaData(session) }
+      metadata = factory.transacter.transaction { session -> loadMetaData(session) }
 
-    val batchPrecomputer = BatchPrecomputer(this)
-    val batchQueuer = BatchQueuer(this, metadata.numThreads)
-    val batchRunner = BatchRunner(
-        this,
-        batchQueuer.nextBatchChannel(),
-        metadata.numThreads
-    )
-    val batchAwaiter = BatchAwaiter(
-        this,
-        batchRunner.runChannel(),
-        batchRunner.rpcBackpressureChannel()
-    )
+      val batchPrecomputer = BatchPrecomputer(this)
+      val batchQueuer = BatchQueuer(this, metadata.numThreads)
+      val batchRunner = BatchRunner(
+          this,
+          batchQueuer.nextBatchChannel(),
+          metadata.numThreads
+      )
+      val batchAwaiter = BatchAwaiter(
+          this,
+          batchRunner.runChannel(),
+          batchRunner.rpcBackpressureChannel()
+      )
 
-    // All our tasks run on this thread.
-    runBlocking {
-      batchPrecomputer.run(this)
-      batchQueuer.run(this)
-      batchRunner.run(this)
-      batchAwaiter.run(this)
+      // All our tasks run on this thread.
+      runBlocking(MDCContext()) {
```

Comment:
hadn't even thought about this, although it shouldn't matter cause we use a single thread coroutine. But if this works great

Reply:
Yeah supposedly this should work.